main.rb

```diff
@@ -0,0 +1,21 @@
+def bubble_sort(item)
+  (1..(item.length - 1)).each do
+    (0..(item.length - 2)).each do |k|
+      item[k], item[k + 1] = item[k + 1], item[k] if item[k].to_i > item[k + 1].to_i
+    end
+  end
+  item = item.inspect
+end
+
+def bubble_sort_by(item)
+  (1..(item.length - 1)).each do
+    (0..(item.length - 2)).each do |k|
+      item[k], item[k + 1] = item[k + 1], item[k] if item[k].length > item[k + 1].length
+    end
+  end
+  item = item.inspect
+end
```

Comment:
* [ ] bubble_sort_by method is implemented and working **(covered by automated tests)** ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1))

`bubble_sort_by(%w[hi hello hey]) { |left, right| left.length - right.length }` returns `"[\"hi\", \"hey\", \"hello\"]"` instead of `["hi", "hey", "hello"]`